src/components/application_manager/src/message_helper/message_helper.cc

```diff
@@ -1434,7 +1434,13 @@ smart_objects::SmartObjectList MessageHelper::CreateAddCommandRequestToHMI(
 
       smart_objects::SmartObject msg_params =
           smart_objects::SmartObject(smart_objects::SmartType_Map);
-      msg_params[strings::cmd_id] = i->first;
+
+      if ((*i->second).keyExists(strings::cmd_id)) {
+        msg_params[strings::cmd_id] = (*i->second)[strings::cmd_id].asUInt();
+      } else {
+        LOG4CXX_ERROR(logger_, "Command ID is missing.");
+        return requests;
```

Comment:
What about the rest of the commands?

```suggestion
        continue;
```

Reply:
@iCollin fix in [641713c](https://github.com/smartdevicelink/sdl_core/pull/2515/commits/641713c8103e6dac7314734aaed2bd714bdf189f)